src/SimplifyCorrelatedDifferences.cpp

```diff
@@ -218,9 +219,15 @@ class SimplifyCorrelatedDifferences : public IRMutator {
             e = PartiallyCancelDifferences().mutate(e);
             e = simplify(e);
 
-            if ((debug::debug_level() > 0) &&
+            const bool check_non_monotonic = debug::debug_level() > 0 || get_telemetry() != nullptr;
+            if (check_non_monotonic &&
                 is_monotonic(e, loop_var) == Monotonic::Unknown) {
                 // Might be a missed simplification opportunity. Log to help improve the simplifier.
+                if (get_telemetry()) {
+                    get_telemetry()->record_non_monotonic_loop_var(loop_var, e);
+                }
+                // TODO: should we remove the debug statement, in favor of
```

Comment:
The debug statements are still useful because they tell you *when* during compilation the non-monotonic expr was found. E.g. just before allocation bounds inference is more interesting than after.

Reply:
Will remove the TODO